Gruntfile.js

```diff
@@ -628,27 +643,30 @@ module.exports = (grunt) => {
     killKeepAlive = this.async()
   })
 
-  grunt.registerTask('test', ['build', 'exec:chelDeployAll', 'backend:launch', 'exec:test', 'cypress'])
-  grunt.registerTask('test:unit', ['backend:launch', 'exec:test'])
+  grunt.registerTask('test', ['build', 'exec:chelDeployAll', 'deno:start', 'exec:test', 'cypress', 'deno:stop', 'flow:stop'])
+  grunt.registerTask('test:unit', ['deno:start', 'exec:test', 'deno:stop'])
 
   // -------------------------------------------------------------------------
   //  Process event handlers
   // -------------------------------------------------------------------------
 
-  process.on('exit', () => {
+  process.on('exit', (code, signal) => {
+    console.log('[node] Exiting with code:', code, 'signal:', signal)
     // Note: 'beforeExit' doesn't work.
     // In cases where 'watch' fails while child (server) is still running
     // we will exit and child will continue running in the background.
     // This can happen, for example, when running two GIS instances via
     // the PORT_SHIFT envar. If grunt-contrib-watch livereload process
     // cannot bind to the port for some reason, then the parent process
     // will exit leaving a dangling child server process.
-    if (child) {
+    if (child && !child.killed) {
       grunt.log.writeln('Quitting dangling child!')
-      child.send({ shutdown: 2 })
+      child.kill()
     }
-    // Stops the Flowtype server.
-    exec('./node_modules/.bin/flow stop')
+    // Make sure to stop the Flowtype server in case `flow:stop` wasn't called.
+    exec('./node_modules/.bin/flow stop', () => {
+      grunt.log.writeln('Flowtype server stopped in process exit handler')
+    })
```

Comment:
Just a note: Sebin recently updated the Gruntfile to make sure that `grunt pin` runs esbuild first. Check https://github.com/okTurtles/group-income/pull/1402/files for what to bring back here